common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/client/LwM2mClientProfile.java

```diff
@@ -51,4 +51,14 @@
      * [ "/2/0/0", "/2/0/1"]
      */
     JsonArray postObserveProfile;
+
+    public LwM2mClientProfile clone () {
+        LwM2mClientProfile lwM2mClientProfile = new LwM2mClientProfile();
+        lwM2mClientProfile.postClientLwM2mSettings =this.postClientLwM2mSettings;
```

Comment:
This clone method is actually misleading. You should clone the JSON objects as well (fields of the object).